mesa/time.py

```diff
@@ -67,19 +67,19 @@ def remove(self, agent):
             agent: An agent object.
 
         """
-        while agent in self.agents:
-            self.agents.remove(agent)
+        del self.agents[agent.unique_id]
 
     def step(self):
         """ Execute the step of all the agents, one at a time. """
-        for agent in self.agents[:]:
-            agent.step()
+        agent_keys = list(self.agents.keys())
```

Comment:
Actually, in all the places where `.values()` are used, the order does not matter since they are for data collection/record or asserting properties. This is the only place where the order matters.
But in one way or another, order preserving still has to be enforced so that users running a Mesa simulation don't accidentally fall into the pitfall -- principle of least surprise.